gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -715,3 +847,56 @@ def __init__(
 
   def expand(self, pvalue):
     return pvalue.pipeline | Read(self._source)
+
+
+class WriteToVcf(PTransform):
+  """A PTransform for writing to VCF files."""
+
+  def __init__(self,
+               file_path_prefix,
+               file_name_suffix='',
+               num_shards=0,
+               shard_name_template=None,
+               coder=coders.ToStringCoder(),
+               compression_type=CompressionTypes.AUTO,
+               header=None):
+    """Initialize a WriteToVcf PTransform.
+
+    Args:
+      file_path_prefix: The file path to write to. The files written will begin
+        with this prefix, followed by a shard identifier (see num_shards), and
+        end in a common extension, if given by file_name_suffix. In most cases,
+        only this argument is specified and num_shards, shard_name_template, and
+        file_name_suffix use default values.
+      file_name_suffix: Suffix for the files written.
+      num_shards: The number of files (shards) used for output. If not set, the
+        service will decide on the optimal number of shards.
+        Constraining the number of shards is likely to reduce
+        the performance of a pipeline.  Setting this value is not recommended
+        unless you require a specific number of output files.
+      shard_name_template: A template string containing placeholders for
+        the shard number and shard count. Currently only '' and
+        '-SSSSS-of-NNNNN' are patterns accepted by the service.
+        When constructing a filename for a particular shard number, the
+        upper-case letters 'S' and 'N' are replaced with the 0-padded shard
+        number and shard count respectively.  This argument can be '' in which
+        case it behaves as if num_shards was set to 1 and only one file will be
+        generated. The default pattern used is '-SSSSS-of-NNNNN'.
+      coder: Coder used to encode each line.
+      compression_type: Used to handle compressed output files. Typical value
+          is CompressionTypes.AUTO, in which case the final file path's
+          extension (as determined by file_path_prefix, file_name_suffix,
+          num_shards and shard_name_template) will be used to detect the
+          compression.
+      header: String to write at beginning of file as a header.
```

Comment:
Did you figure out how to handle sample names? Should they be passed in as a separate argument or can we generate them within the pipeline?

Reply:
In beam, I think the best way to deal with sample names is to have the user create their own header and pass it in through this parameter. Since we are not sorting the samples at all, if they are working with a single file, the last header line should be the same as the original file.